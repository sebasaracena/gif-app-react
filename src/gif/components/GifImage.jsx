import Proptypes from 'prop-types';

export const GifImage=({ title, url })=>{
    return (<>
    <img src={ url } alt={ title } className="img-fluid" />
        <p>{ title }</p>
    </>);
}


GifImage.Proptypes={
   title: Proptypes.string.isRequired,
   url: Proptypes.string.isRequired   

}