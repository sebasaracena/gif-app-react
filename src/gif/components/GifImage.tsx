export const GifImage=({ title, url }:any)=>{
    return (<>
    <img src={ url } alt={ title } className="img-fluid" />
        <p>{ title }</p>
    </>);
}