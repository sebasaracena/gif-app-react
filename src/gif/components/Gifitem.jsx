
import Card from 'react-bootstrap/Card';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifImage } from './index';
import Pagination from 'react-bootstrap/Pagination';





export const Gifitem=({item})=>{
  let active = 2;
  let items = [];
  const { images, isLoading  } = useFetchGifs(item,0);
 
  

  for (let number = 1; number <= 12; number++) {
    items.push(
      <Pagination.Item key={number} active={number === active}>
        {number}
      </Pagination.Item>,
    );
  }

    return (<>
    
     <Card style={{ width: '100%', marginBottom: '10px'}}>  
           <Card.Body>
             <Card.Title>{item}</Card.Title><hr/>
             {
                isLoading && ( <div><h2>Cargando...</h2></div> )
            }
             {images.map( ( image ) => (
            <GifImage key={image.id} { ...image }/>
             ))}
             <div className='d-flex  justify-content-center mt-4'>
            
              {/*<Pagination>{items}</Pagination>*/}
            </div>
           </Card.Body>
         </Card>
    </>);
}