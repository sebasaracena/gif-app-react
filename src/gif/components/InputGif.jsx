import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import InputGroup from 'react-bootstrap/InputGroup';
import { useState } from 'react';




export const InputGif=({onNewCategory})=>{
    const [inputCategorie,setInputCategorie]=useState('');

    const onInputChange = ({ target } ) => {
        setInputCategorie( target.value );
    }

    
    const onSubmit = ( event ) => {
        
        event.preventDefault();
        if( inputCategorie.trim().length <= 1) return;

        setInputCategorie('');
        onNewCategory( inputCategorie.trim() );
    }

    return (<>
      <InputGroup className="mb-3">
        <Form.Control
          placeholder="Buscar categoria"
          value={inputCategorie}
          aria-label="Recipient's username"
          aria-describedby="basic-addon2"
          onChange={ onInputChange }
        />
        <Button variant="outline-primary" id="button-addon2" onClick={onSubmit}>
        buscar
        </Button>
      </InputGroup>
    </>);
}

