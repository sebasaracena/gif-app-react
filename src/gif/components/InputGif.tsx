import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import InputGroup from 'react-bootstrap/InputGroup';
import { useState } from 'react';




export const InputGif=({onNewCategory}:any)=>{
    const [inputCategorie,setInputCategorie]=useState('');

    const onInputChange = ({ target }:any) => {
        setInputCategorie( target.value );
    }

    
    const onSubmit = ( event:any ) => {
        
        event.preventDefault();
        if( inputCategorie.trim().length <= 1) return;

        setInputCategorie('');
        onNewCategory( inputCategorie.trim() );
    }

    return (<>
      <InputGroup className="mb-3">
        <Form.Control
          placeholder="Buscar categoria"
          value={inputCategorie}
          aria-label="Recipient's username"
          aria-describedby="basic-addon2"
          onChange={ onInputChange }
        />
        <Button variant="outline-primary" id="button-addon2" onClick={onSubmit}>
        buscar
        </Button>
      </InputGroup>
    </>);
}

