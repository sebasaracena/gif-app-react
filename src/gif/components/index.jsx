export * from './Gifitem';
export * from './GifImage';
export * from './InputGif';