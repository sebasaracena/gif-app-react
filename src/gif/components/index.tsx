export * from './Gifitem';
export * from './GifImage';
export * from '../../GifCategorie';
export * from './InputGif';