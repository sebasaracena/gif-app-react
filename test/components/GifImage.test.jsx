
import { render, screen } from '@testing-library/react';
import { GifImage  } from '../../src/gif/components/GifImage';

describe('Pruebas en <GifImage />', () => {

    const title = 'Saitama';
    const url   = 'https://one-punch.com/saitama.jpg';

    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render( <GifImage title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
        render( <GifImage title={ title } url={ url } /> );
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    });


    test('debe de mostrar el titulo en el componente', () => {
        
        render( <GifImage title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();

    });
    
});