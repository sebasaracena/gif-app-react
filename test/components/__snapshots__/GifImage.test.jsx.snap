// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifImage /> debe de hacer match con el snapshot 1`] = `
<div>
  <img
    alt="Saitama"
    class="img-fluid"
    src="https://one-punch.com/saitama.jpg"
  />
  <p>
    Saitama
  </p>
</div>
`;
